### PR TITLE
Implementation of the index design

### LINKED ISSUE #6: HTML and SASS Implementation [index page]
# REQUIREMENTS
Must have the initial files of HTML and SASS to the index page following the design.

# DESIGN
image

### PR DESCRIPTION
**Pull request by germanruzca**
# Description
What: The design from mockup was implemented (header and main page), it was used scss framework to the html files, also use inputs, buttons and labels, addition of scss components, use of global variables and Font Aweosme and Google Fonts tools. 
Why: With these changes is finished the main page view, to the user/employee.

See issue #6 for more background.
 ## Screenshots
image

### CODE REVIEW

File: app/assets/stylesheets/base/_globals.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+h1, h2, h3 {
+  font-family: 'Roboto', sans-serif;
+  font-weight: 700;
+}
+
+p {
+  font-family: 'Roboto', sans-serif;
+  font-weight: 300;
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/base/_globals.scss]: Add a blank line here

File: app/assets/stylesheets/components/_buttons.scss [hunk 2]
```diff
@@ -0,0 +1,9 @@
+button {
+  border-radius: 10px;
+  border: none;
+  font-family: $font;
+  font-size: 20px;
+  height: 50px;
+  width: 200px;
+  cursor: pointer;
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_buttons.scss]: sort this block

File: app/assets/stylesheets/components/_buttons.scss [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_buttons.scss]: Add blank line here

File: app/assets/stylesheets/layout/_header.scss [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    transform: translateX(-50%);
+  }
+  
+  .primary-button-style {
+    background-color: $red_button;
+    color: $wheat_button;
+    transition: all .4s ease-in-out;
+    font-weight: 700;
+    flex: 0 1 auto;
+    margin-left: auto;
```
Comment (JuanequeX) [inline app/assets/stylesheets/layout/_header.scss]: sort this block

File: app/assets/stylesheets/layout/_header.scss [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  align-items: center;
+  
+  .title {
+    color: $white_title;
+    font-size: 70px;
+    align-self: auto;
+    flex: 0 1 auto;
+    position: absolute;
+    left: 50%;
+    transform: translateX(-50%);
```
Comment (JuanequeX) [inline app/assets/stylesheets/layout/_header.scss]: Sort this block

File: app/assets/stylesheets/layout/_header.scss [hunk 6]
```diff
@@ -0,0 +1,34 @@
+.header {
+  background-color: $blue_background;
+  box-sizing: border-box;
+  display: flex;
+  height: 120px;
+  justify-content: flex-start; 
+  position: relative;  
+  padding: 0 20px;
+  width: 100%;
+  align-items: center;
```
Comment (JuanequeX) [inline app/assets/stylesheets/layout/_header.scss]: Sort this block

File: app/assets/stylesheets/layout/_header.scss [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    transition: all .4s ease-in-out;
+    font-weight: 700;
+    flex: 0 1 auto;
+    margin-left: auto;      
+    
+    &:hover {
+      background-color: $red_button_hover;
+    }
+  }
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/layout/_header.scss]: Add blank line here

File: app/assets/stylesheets/pages/_index.scss [hunk 8]
```diff
@@ -0,0 +1,75 @@
+.main {
+  background-color: $blue_background;
+  height: 85vh;
+  display: flex;
+  justify-content: space-around;
```
Comment (JuanequeX) [inline app/assets/stylesheets/pages/_index.scss]: Sort this block

File: app/assets/stylesheets/pages/_index.scss [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+  .time {
+    align-items: center;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    margin: 0;
+    font-family: $font;
+
```
Comment (JuanequeX) [inline app/assets/stylesheets/pages/_index.scss]: Sort this block

File: app/assets/stylesheets/pages/_index.scss [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+    align-items: center;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    margin: 0;
+    font-family: $font;
+  
+    .hour {
+      font-size: 300px;
+      color:  $black_hour_color;
```
Comment (JuanequeX) [inline app/assets/stylesheets/pages/_index.scss]: Sort this block

File: app/assets/stylesheets/pages/_index.scss [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+      .type {
+        font-size: 40px;
+        flex-wrap: nowrap;
+      }
+    }
+
+    .date {
+      font-size: 60px;
+      color: $black_hour_color;
```
Comment (JuanequeX) [inline app/assets/stylesheets/pages/_index.scss]: Sort this block

File: app/assets/stylesheets/pages/_index.scss [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+  .number {
+    display: flex;
+    flex-direction: column;
+    padding: 180px 0;
+
+    h3 {
+      font-size: 45px;
+      margin-bottom: 10px;
+      font-family: $font;
+      color: $aqua_label;
```
Comment (JuanequeX) [inline app/assets/stylesheets/pages/_index.scss]: Sort this block

File: app/assets/stylesheets/pages/_index.scss [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    input {
+      width: 350px;
+      height: 50px;
+      font-size: 40px;
+      font-family: $font;
+      border: 0;
+      border-radius: 10px;
+      padding: 10px;
```
Comment (JuanequeX) [inline app/assets/stylesheets/pages/_index.scss]: Sort this block

File: app/assets/stylesheets/pages/_index.scss [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+      opacity: 0.2;
+    }
+
+    button {
+      width: 200px;
+      height: 40px;
+      margin: 20px;
+      font-family: $font;
+      font-size: 30px;
+      align-self: center;
```
Comment (JuanequeX) [inline app/assets/stylesheets/pages/_index.scss]: Sort this block

File: app/assets/stylesheets/pages/_index.scss [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+    }
+  }
+  
+  .secondary-button-style {
+      background-color: $red_button;
+      color: $wheat_button;
+      font-weight: 700;
```
Comment (JuanequeX) [inline app/assets/stylesheets/pages/_index.scss:69]: Sort this block

File: app/views/home/index.html.erb [hunk 16]
```diff
@@ -1,2 +1,19 @@
...
+  <div class="number">
+    <h3>
+      Private number
+    </h3>
+    <input type="text" class="input-text" placeholder="Enter number">
+    <button class="secondary-button-style">
+      CHECK
+    </button>
+  </div>
+</div>
```
Comment (JuanequeX) [inline app/views/home/index.html.erb]: Add a blank line here

File: app/views/layouts/_header.html.erb [hunk 17]
```diff
@@ -0,0 +1,6 @@
+<div class="header">
+  <h1 class="title">Check In <i class="fa-solid fa-clock"></i></h1>
+  <button class="primary-button-style">
+    Admin Tools
+  </button>
+</div>
```
Comment (JuanequeX) [inline app/views/layouts/_header.html.erb]: Add a blank line here

**Review by JuanequeX — CHANGES_REQUESTED** [review]

File: app/views/layouts/application.html.erb [hunk 18]
```diff
@@ -1,16 +1,19 @@
...
   <head>
-    <title>Kata07RelojChecadorK7F22T2</title>
+    <title>Check Clock</title>
     <meta name="viewport" content="width=device-width,initial-scale=1">
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
     <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>
     <%= javascript_importmap_tags %>
+    <link rel="stylesheet" href="path/to/font-awesome/css/font-awesome.min.css">
```
Comment (YilKanda) [inline app/views/layouts/application.html.erb]: Is this line of code functional?

File: app/assets/stylesheets/base/_globals.scss (hunk 1 above)
Comment (germanruzca) [reply app/assets/stylesheets/base/_globals.scss]: Done

File: app/assets/stylesheets/components/_buttons.scss (hunk 2 above)
Comment (germanruzca) [reply app/assets/stylesheets/components/_buttons.scss]: Done

File: app/assets/stylesheets/components/_buttons.scss (hunk 3 above)
Comment (germanruzca) [reply app/assets/stylesheets/components/_buttons.scss]: Done

File: app/assets/stylesheets/layout/_header.scss (hunk 4 above)
Comment (germanruzca) [reply app/assets/stylesheets/layout/_header.scss]: Done

File: app/assets/stylesheets/layout/_header.scss (hunk 5 above)
Comment (germanruzca) [reply app/assets/stylesheets/layout/_header.scss]: Done

File: app/assets/stylesheets/layout/_header.scss (hunk 6 above)
Comment (germanruzca) [reply app/assets/stylesheets/layout/_header.scss]: Done

File: app/assets/stylesheets/layout/_header.scss (hunk 7 above)
Comment (germanruzca) [reply app/assets/stylesheets/layout/_header.scss]: Done

File: app/assets/stylesheets/pages/_index.scss (hunk 8 above)
Comment (germanruzca) [reply app/assets/stylesheets/pages/_index.scss]: Done

File: app/assets/stylesheets/pages/_index.scss (hunk 9 above)
Comment (germanruzca) [reply app/assets/stylesheets/pages/_index.scss]: Done

File: app/assets/stylesheets/pages/_index.scss (hunk 10 above)
Comment (germanruzca) [reply app/assets/stylesheets/pages/_index.scss]: Done

File: app/assets/stylesheets/pages/_index.scss (hunk 11 above)
Comment (germanruzca) [reply app/assets/stylesheets/pages/_index.scss]: Done

File: app/assets/stylesheets/pages/_index.scss (hunk 12 above)
Comment (germanruzca) [reply app/assets/stylesheets/pages/_index.scss]: Done

File: app/assets/stylesheets/pages/_index.scss (hunk 13 above)
Comment (germanruzca) [reply app/assets/stylesheets/pages/_index.scss]: Done

File: app/assets/stylesheets/pages/_index.scss (hunk 14 above)
Comment (germanruzca) [reply app/assets/stylesheets/pages/_index.scss]: Done

File: app/assets/stylesheets/pages/_index.scss (hunk 15 above)
Comment (germanruzca) [reply app/assets/stylesheets/pages/_index.scss:69]: This block is already sort

File: app/views/home/index.html.erb (hunk 16 above)
Comment (germanruzca) [reply app/views/home/index.html.erb]: Done

File: app/views/layouts/_header.html.erb (hunk 17 above)
Comment (germanruzca) [reply app/views/layouts/_header.html.erb]: Done

File: app/views/layouts/application.html.erb (hunk 18 above)
Comment (germanruzca) [reply app/views/layouts/application.html.erb]: I just checked, is not needed, it was removed. Thank you!

**Review by JuanequeX — APPROVED** [review]
LGTM

**Review by juperameza — APPROVED** [review]
LGTM :) ♥

**Review by YilKanda — APPROVED** [review]
LGTM!

**Review by vyk2rr — APPROVED** [review]

**Pull request merged**